stravalib/client.py

```diff
@@ -1903,9 +1903,15 @@ def _fill_buffer(self):
 
         entities = []
         for raw in raw_results:
-            entities.append(
-                self.entity.deserialize(raw, bind_client=self.bind_client)
-            )
+            try:
+                new_entity = self.entity.parse_obj(raw)
+                # TODO: try adding bind_client to entity
+            except AttributeError:
+                # entity doesn't have a parse_obj() method, so must be of a legacy type
+                new_entity = self.entity.deserialize(
+                    raw, bind_client=self.bind_client
+                )
+            entities.append(new_entity)
```

Comment:
This makes sure that the `BatchedResultsIterator` can parse the new pydantic-based entities (but has a temporary workaround to also handle legacy entities).

Reply:
Ok question about the activity results. 

i'm seeing a lot of information in the results that is empty 

```python
# Get activities
act = client.get_activities(limit=12)
for a in act:
    print(a)

a_dict = a.to_dict()
a_dict.keys()
a_dict["athlete"]
```
OUTPUT
```
{'id': None, 'city': None, 'country': None, 'created_at': None, 'firstname': None, 'lastname': None, 'premium': None, 'profile': None, 'profile_medium': None, 'resource_state': None, 'sex': None, 'state': None, 'summit': None, 'updated_at': None, 'bikes': None, 'clubs': None, 'follower_count': None, 'friend_count': None, 'ftp': None, 'measurement_preference': None, 'shoes': None, 'weight': None}
```

i'm just curious why some of this is in the return for activities as it's in the athlete data but that doesn't need to be returned for every activity as it would make the return object quite large.

The other odd thing is the return of a single activit also has this::
```
a.ALPINESKI                    a.embed_token
a.BACKCOUNTRYSKI               a.end_latlng
a.CANOEING                     a.expand(
a.CROSSCOUNTRYSKIING           a.external_id
a.CROSSFIT                     a.flagged
a.EBIKERIDE                    a.from_accepted_tag
a.ELLIPTICAL                   a.from_dict(
a.GOLF                         a.full_photos
```

if the activity has a single type why does it list all of the possible types in the return? i am just curious about what is in this return and why there is so much extra stuff associated with a single activity. (thinking about people processing thousands of activities).

otherwise the code is all working as expected i'm just getting some unexpected return VALUES.

Reply:
What you see for activities is still the legacy API, as I haven't replaced the `Activity` entity yet. That list of _possible_ activity types is probably some legacy way of supporting enum-like behavior (python didn't support that before 3.4). The generated pydantic model now has a proper enum, so these will not appear in an `Activity` object.

Reply:
> i'm just curious why some of this is in the return for activities as it's in the athlete data but that doesn't need to be returned for every activity as it would make the return object quite large

Also this is just part of the legacy API, since it only supports one `Athlete` class (whereas the pydantic model you can see in `strava_model.py` has distinct types such as `MetaAthlete`, `SummaryAthlete` and `DetailedAthlete`. We can later add issues to extend these other types to limit the size of (nested) objects (I've noted this down).

Reply:
Ok that all makes sense. I understand this is a very big work in progress!! 

I really like testing this in small pieces so let's add those as issues later if we need to limit redundancy of info in the returned nest data. Many thanks for clarifying.